src/mono/mono/metadata/native-library.c

```diff
@@ -11,13 +11,22 @@
 #include "mono/utils/mono-logger-internals.h"
 #include "mono/utils/mono-path.h"
 #include "mono/metadata/native-library.h"
+#include "mono/metadata/custom-attrs-internals.h"
 
 #ifdef ENABLE_NETCORE
 static int pinvoke_search_directories_count;
 static char **pinvoke_search_directories;
 
-// In DllImportSearchPath enum, bit 0x2 represents AssemblyDirectory. It is not passed on and is instead handled by the runtime.
-#define DLLIMPORTSEARCHPATH_ASSEMBLYDIRECTORY 0x2
+typedef enum
+{
+	DLLIMPORTSEARCHPATH_LegacyBehavior = 0x0,
+	DLLIMPORTSEARCHPATH_UseDllDirectoryForDependencies = 0x100,
+	DLLIMPORTSEARCHPATH_ApplicationDirectory = 0x200,
+	DLLIMPORTSEARCHPATH_UserDirectories = 0x400,
+	DLLIMPORTSEARCHPATH_System32 = 0x800,
+	DLLIMPORTSEARCHPATH_SafeDirectories = 0x1000,
+	DLLIMPORTSEARCHPATH_AssemblyDirectory = 0x2,
+} DLLIMPORTSEARCHPATH;
```

Comment:
We usually use `ALL_CAPS` with underscores between words (`DLL_IMPORT_SEARCH_PATH_LEGACY`) for the enum values and `MixedCase` for the enum type.

Also this needs a comment to go look at the corresponding managed enum.